libdnf/dnf-context.h

```diff
@@ -311,11 +311,12 @@ gboolean         dnf_context_module_enable              (DnfContext * context,
  *
  * Enable modules and mark for installation but do not commit modular changes.
  * To commit modular changes, call dnf_context_run().
- * Returns FALSE when an error is set.
+ * Returns the list of fully resolved modules marked for installation in NSVCA/P
+ * form or NULL when an error is set.
  *
  * Since: 0.63.0
  **/
-gboolean         dnf_context_module_install              (DnfContext * context,
+char**           dnf_context_module_install              (DnfContext * context,
```

Comment:
Be honest - such a change is always problematic. It requires change in so name version or provide the change as a new function to keep compatibility. It also makes a difference to other functions that handles modularity.

Reply:
I also think that the API change will be misleading. Modules are not installed but profiles. Also packages are not installed only from the latest modules but also nonmodular packages or packages from other module streams. May be the output from `getInstalledProfiles();` can be sufficient.

Reply:
> Be honest - such a change is always problematic. It requires change in so name version or provide the change as a new function to keep compatibility. It also makes a difference to other functions that handles modularity.

When I initially opened this PR, there were no releases yet with the new API so it was safe to change it. But with the API out in v0.63.0, that's no longer the case. So agreed, this is not safe to do.

I'll add a separate API for this.

Reply:
@jlebon The most important questions is: Do you really need it? Probably there is a way how you can get what do you need without the change of API. Please could you explain me, why do you need it? Or how exactly you will use the returned strings?

Reply:
I'd really like to include the fully resolved NSVCA/P in the OSTree metadata just on principle so that it's inspectable. Otherwise, because modules purely live in yumrepo metadata and not e.g. on disk, there isn't really any trace of what exact modules and their versions are part of an OSTree commit.

But let me see, I think we might be able to do without it and just store the user-provided string as is. The biggest concern there is handling globby patterns like `postgresq?:13/client` (obviously not a realistic example), but maybe better anyway to just deny globby patterns and force users to type out the module spec more explicitly.

Reply:
`postgresq?:13/client` is nice and easy to hadle example, but `nodejs:*` is more interesting example. <spec> matches multiple streams.

Reply:
OK, dropped the need for this now and just went with the naive approach of storing the user-provided string as is. At least the modular RPMs are still in the rpmdb, so one can sort of infer what modules were installed. We'll look at adding this in a follow-up if we really need it.